warehouse/ingest-ssdeep/src/main/java/datawave/ingest/mapreduce/handler/ssdeep/SSDeepIndexHandler.java

```diff
@@ -0,0 +1,238 @@
+
+    public static final String SSDEEP_BUCKET_COUNT = ".ssdeepIndex.bucket.count";
+
+    public static final int DEFAULT_SSDEEP_BUCKET_COUNT = BucketAccumuloKeyGenerator.DEFAULT_BUCKET_COUNT;
+
+    public static final String SSDEEP_BUCKET_ENCODING_BASE = ".ssdeepIndex.bucket.encoding.base";
+
+    public static final int DEFAULT_SSDEEP_BUCKET_ENCODING_BASE = BucketAccumuloKeyGenerator.DEFAULT_BUCKET_ENCODING_BASE;
+
+    public static final String SSDEEP_BUCKET_ENCODING_LENGTH = ".ssdeepIndex.bucket.encoding.length";
+
+    public static final int DEFAULT_SSDEEP_BUCKET_ENCODING_LENGTH = BucketAccumuloKeyGenerator.DEFAULT_BUCKET_ENCODING_LENGTH;
+
+    public static final String SSDEEP_INDEX_NGRAM_SIZE = ".ssdeepIndex.ngram.size.min";
+
+    public static final int DEFAULT_SSDEEP_INDEX_NGRAM_SIZE = 7;
+
+    public static final String SSDEEP_MIN_HASH_SIZE = ".ssdeepIndex.chunk.size.min";
+
+    public static final int DEFAULT_SSDEEP_MIN_HASH_SIZE = 3;
+
+    protected Text ssdeepIndexTableName;
+
+    protected MarkingFunctions markingFunctions;
+
+    protected TaskAttemptContext taskAttemptContext;
+
+    protected Set<String> ssdeepFieldNames;
+
+    protected NGramByteHashGenerator nGramGenerator;
+    protected BucketAccumuloKeyGenerator accumuloKeyGenerator;
+
+    @Override
+    public void setup(TaskAttemptContext context) {
+        markingFunctions = MarkingFunctions.Factory.createMarkingFunctions();
+        taskAttemptContext = context;
+
+        Configuration conf = context.getConfiguration();
+
+        final String t = ConfigurationHelper.isNull(conf, DataTypeHelper.Properties.DATA_NAME, String.class);
+        TypeRegistry.getInstance(conf);
+        Type type = TypeRegistry.getType(t);
+
+        int bucketCount = conf.getInt(type.typeName() + SSDEEP_BUCKET_COUNT, DEFAULT_SSDEEP_BUCKET_COUNT);
+        int bucketEncodingBase = conf.getInt(type.typeName() + SSDEEP_BUCKET_ENCODING_BASE, DEFAULT_SSDEEP_BUCKET_ENCODING_BASE);
+        int bucketEncodingLength = conf.getInt(type.typeName() + SSDEEP_BUCKET_ENCODING_LENGTH, DEFAULT_SSDEEP_BUCKET_ENCODING_LENGTH);
+        int ngramSize = conf.getInt(type.typeName() + SSDEEP_INDEX_NGRAM_SIZE, DEFAULT_SSDEEP_INDEX_NGRAM_SIZE);
+        int minHashSize = conf.getInt(type.typeName() + SSDEEP_MIN_HASH_SIZE, DEFAULT_SSDEEP_MIN_HASH_SIZE);
+
+        accumuloKeyGenerator = new BucketAccumuloKeyGenerator(bucketCount, bucketEncodingBase, bucketEncodingLength);
+        nGramGenerator = new NGramByteHashGenerator(ngramSize, bucketCount, minHashSize);
+
+        String[] fieldNameArray = conf.getStrings(type.typeName() + SSDEEP_FIELD_SET);
+        ssdeepFieldNames = new HashSet<>(List.of(fieldNameArray));
+        ssdeepIndexTableName = new Text(ConfigurationHelper.isNull(conf, SSDEEP_INDEX_TABLE_NAME, String.class));
+    }
+
+    @Override
+    public String[] getTableNames(Configuration conf) {
+        return getNonNullTableNames(conf, SSDEEP_INDEX_TABLE_NAME);
```

Comment:
This seems like a horribly complicated way to return new String[] {ssdeepIndexTableName};

Reply:
Yes, was really just following the pattern from elsewhere (probably the Facet handler or other index handlers).

Reply:
We should clean those up.... sigh.